Source/options.cpp

```diff
@@ -572,6 +568,90 @@ std::vector<OptionEntryBase *> AudioOptions::GetEntries()
 	};
 }
 
+OptionEntryResolution::OptionEntryResolution()
+    : OptionEntryListBase("", OptionEntryFlags::CantChangeInGame | OptionEntryFlags::RecreateUI, N_("Resolution"), N_("Affect the game's internal resolution and determine your view area. Note: This can differ from screen resolution, when Upscaling, Integer Scaling or Fit to Screen is used."))
+{
+}
+void OptionEntryResolution::LoadFromIni(string_view category)
+{
+	size = { GetIniInt(category.data(), "Width", DEFAULT_WIDTH), GetIniInt(category.data(), "Height", DEFAULT_HEIGHT) };
+}
+void OptionEntryResolution::SaveToIni(string_view category) const
+{
+	SetIniValue(category.data(), "Width", size.width);
+	SetIniValue(category.data(), "Height", size.height);
+}
+
+void OptionEntryResolution::CheckResolutionsAreInitialized() const
+{
+	if (!resolutions.empty())
+		return;
+
+	std::vector<Size> sizes;
+
+	// Add resolutions
+#ifdef USE_SDL1
+	auto *modes = SDL_ListModes(nullptr, SDL_FULLSCREEN | SDL_HWPALETTE);
+	if (modes != nullptr) {
+		for (size_t i = 0; modes[i] != nullptr; i++) {
+			sizes.emplace_back(Size { modes[i]->w, modes[i]->h });
+		}
+	}
+#else
+	int displayModeCount = SDL_GetNumDisplayModes(0);
+	for (int i = 0; i < displayModeCount; i++) {
+		SDL_DisplayMode mode;
+		if (SDL_GetDisplayMode(0, i, &mode) != 0) {
+			ErrSdl();
+		}
+		sizes.emplace_back(Size { mode.w, mode.h });
+	}
+#endif
+
+	// Ensures that the ini specified resolution is present in resolution list even if it doesn't match a monitor resolution (for example if played in window mode)
+	sizes.push_back(this->size);
+	// Ensures that the vanilla/default resolution is always present
+	sizes.emplace_back(Size { DEFAULT_WIDTH, DEFAULT_HEIGHT });
+
+	// Sort by width then by height

```

Comment:
Might be better to do it by height first since most designated resolutions by height alone (480p 720p 1080p 1440p etc)

Reply:
I tried to mimic how my os (windows 10) and starcraft 2 presents me the resolution list:
![grafik](https://user-images.githubusercontent.com/25415264/144931198-a4fb1c1f-30dc-4d94-bc55-b011a70ce93c.png) ![grafik](https://user-images.githubusercontent.com/25415264/144931568-ba8eaa9d-62ae-4cb9-b84b-09fea0860c1e.png)
There it is sorted by width first and then by height.

But I don't have a strong opinion on this, if you want I can change it. 🙂

Reply:
Damn now I want to play StarCraft, good thing it works properly via Wine do I don't have to RE and port it first :D

Reply:
I guess if it was me designing the world i would group them by aspect as well